### ISSUE TITLE
Investigate: tests

### ISSUE THREAD

**Issue by falkben** [spacetelescope/DRESSCode#2]
Since the pipeline is mainly calling into heasoft with subprocess calls, and these are somewhat long running calls, and require a long install process, what and how can we test our pipeline?

We should come up with a plan for
- what needs tests
- how can we test

**Comment by falkben** [comment]
before we make significant changes to the pipeline, we should try to address this -- we want to have some assurances that the pipeline at least runs with some sample data (even if it's just a single file)

**Comment by mdecleir** [comment]
I agree some basic testing would be useful! Testing with a single file might be difficult, because it will likely skip some steps (e.g. the co-adding process) if there is only 1 file. I think a limited number of input files will be needed.

If we want to test whether the entire pipeline runs from A to Z without problems, we will probably first need a script that runs all the steps one after the other, instead of the current manual step-by-step running of scripts in the terminal...

**Comment by falkben** [comment]
> If we want to test whether the entire pipeline runs from A to Z without problems, we will probably first need a script that runs all the steps one after the other, instead of the current manual step-by-step running of scripts in the terminal...

yup, working on this :)

**Comment by mdecleir** [comment]
From the raw images, this could serve as a testing data set (for NGC0628):

sw00032891001um2_rw.img
sw00032891001uw1w1po_uf.evt
sw00032891001uw2_rw.img
sw00032891019um2_rw.img
sw00032891019um2w1po_uf.evt
sw00032891019uw1_rw.img
sw00032891019uw2_rw.img
sw00035868001um2_rw.img
sw00035868001uw1_rw.img
sw00035868001uw2_rw.img
sw00036568001um2_rw.img
sw00036568001uw1_rw.img
sw00036568001uw2_rw.img
sw00036568002uw1_rw.img
sw00036568003uw2_rw.img

It has both images and event tables, from all 3 filters and from all observing years for this galaxy (2007, 2008, 2013 and 2015). This should test all parts of the pipeline.

**Comment by falkben** [comment]
so i needed some fits files as well.

I may have gone overboard. This is what is in my `Raw_images` folder now:

```
sw00032891001pat.fits
sw00032891001uaf.hk
sw00032891001um2_rw.img
sw00032891001uw1_rw.img
sw00032891001uw1w1po_uf.evt
sw00032891001uw2_rw.img
sw00032891019pat.fits
sw00032891019uaf.hk
sw00032891019um2_rw.img
sw00032891019um2w1po_uf.evt
sw00032891019uw1_rw.img
sw00032891019uw2_rw.img
sw00035868001pat.fits
sw00035868001uaf.hk
sw00035868001um2_rw.img
sw00035868001uw1_rw.img
sw00035868001uw2_rw.img
sw00036568001um2_rw.img
sw00036568001uw1_rw.img
sw00036568001uw2_rw.img
sw00036568002pat.fits
sw00036568002uaf.hk
sw00036568002uw1_rw.img
sw00036568003pat.fits
sw00036568003uaf.hk
sw00036568003uw2_rw.img
```

**Comment by mdecleir** [comment]
Ah of course, I forgot the attitude and aspect following files. You will have to add those for sw00036568001 as well.

**Comment by falkben** [comment]
When #21 is merged, we will have a dockerfile we can use to test the latest version of the pipeline within a github action.

Remaining items for this ticket:

1. script to download test data
2. script to run the pipeline from start to finish
3. a github action that pulls the docker image, and runs the two scripts above
4. some minimal check to make sure the pipeline actually executed (maybe just check exit codes from above scripts and presence of output files)

**Comment by mdecleir** [comment]
I revised the minimal test data set, to reduce running time, but still making sure that all functionality is being tested. The list below includes (for future reference):
- at least 1 file from every filter, testing the filter capabilities,
- at least 1 image file and 1 event file with the same observation ID, testing the processing of both file types and how they are combined,
- at least 1 image file with more than 1 frame, testing the handling of different frames within 1 image file,
- at least 1 image file for which at least 1 event based image frame will be skipped during the uvotimage.py and uvotimage2.py steps, testing the skipping of duplicate frames,
- at least 1 frame for which no aspect corrections have been found with uvotskycorr.py, testing throwing a warning when that happens,
- at least 1 frame for which no aspect corrections have been found with uvotskycorr2.py, testing that these frames will be skipped when co-adding the frames,
- at least 1 year with at least 2 files, testing the co-adding of files within the same observing period,
- files from at least 2 different years, testing the combination of the different observing periods,


`sw00032891004pat.fits		sw00032891012uw1_rw.img
sw00032891004um2_rw.img		sw00032891012uw1w1po_uf.evt
sw00032891004um2w1po_uf.evt	sw00032891012uw2_rw.img
sw00032891004uw2_rw.img		sw00032891012uw2w1po_uf.evt
sw00032891004uw2w1po_uf.evt	sw00035868001pat.fits
sw00032891012pat.fits		sw00035868001uaf.hk
sw00032891012uaf.hk		sw00035868001um2_rw.img
sw00032891012um2_rw.img		sw00035868001uw1_rw.img
sw00032891012um2w1po_uf.evt	sw00035868001uw2_rw.img`
